src/test/java/org/example/BDDTesting/stepsDefinitions/LoginFunctionalitiesSteps.java

```diff
@@ -0,0 +1,65 @@
+package org.example.BDDTesting.stepsDefinitions;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+import org.example.BaseTests;
+import org.example.pages.HomePage;
+import org.example.pages.LoginPage;
+
+import io.cucumber.java.en.*;
+
+/**
+ * This class contains the step definitions for the login functionalities.
+ * Each method corresponds to a step in the Cucumber scenarios.
+ */
+public class LoginFunctionalitiesSteps extends BaseTests {
+	private LoginPage loginPage;
+	private HomePage homePage;
+
+	/**
+	 * This method is run when the user navigates to the login page.
+	 */
+	@Given("the user has navigated to the login page")
+	public void theUserHasNavigatedToTheLoginPage() {
+		homePage = new HomePage(driver, logger);
```

Comment:
I guess it can be removed from here, as you have this object already created in BaseTest class, just use it.

Reply:
SetUp Method called from BaseTest Class